compiler/qsc_frontend/src/resolve.rs

```diff
@@ -360,13 +360,22 @@ impl Resolver {
     }
 
     fn bind_type_parameters(&mut self, decl: &CallableDecl) {
+        // ToDo: This seems like the wrong way to get the item_id
+        let item_id = if let Some(Res::Item(item_id, _)) = self.names.get(decl.name.id) {
+            *item_id
+        } else {
+            panic!("Callable decl expected to have item_id in names map.");
+        };
+
```

Comment:
You can simplify this a bit by using the let-else pattern:

```suggestion
        let Some(Res::Item(item_id, _)) = self.names.get(decl.name.id).copied() else {
            panic!("Callable decl expected to have item_id in names map.");
        };

```

Reply:
done